tests/playwright/src/model/pages/volumes-page.ts

```diff
@@ -76,4 +94,26 @@ export class VolumesPage extends MainPage {
     await waitWhile(async () => await this.volumeExists(name));
     return true;
   }
+
+  //returns true if the number of volumes changes or certain time has passed
+  async waitForPruneVolumes(): Promise<boolean> {
+    const previousVolumes = await this.countVolumesFromTable();
+    let currentVolumes;
+    try {
+      await waitWhile(async () => {
+        currentVolumes = await this.countVolumesFromTable();
+        return previousVolumes === currentVolumes;
+      });
+    } catch (error) {
+      return true;
+    }
+    return true;
+  }
```

Comment:
I think I know what you want to do with this, but I don't see how this implementation works.

1. You are calling this method after the prune command was already issues, so by the time it executes some/all volumes might already be pruned, so there is an inate race condition involved.
2. You are awaiting while previousVolume === currentVolume, but what happens if you have many volumes and when you issue the prune command the first ones will disaapear(being deleted) but not yet all, it's possible that previousVolumes and currentVolumes might no longer be equal but that the prune command isn't finished either.
3. I also don't understand why you need to use try catch for waitWhile, the waitWhie method accepts param to not throw error at all.

Reply:
I think I came up with a solution to some of the problems you pointed out, but my guess is that I'm going to have to change the approach I take to this.
```
async waitForPruneVolumes(previousVolumes: number): Promise<boolean> {
    let currentVolumes;
    await waitWhile(async () => {
      currentVolumes = await this.countVolumesFromTable();
      return previousVolumes === currentVolumes;
    }, {sendError: false});
    return true;
  }
```
1. If I pass the previous number of volumes as a parameter (it is calculated before starting the prune always) the race condition should be avoided, as the method only counts the new existing volumes.
2. From my testing, after pruning the volumes the user interface does not refresh until it is done removing them, so I did not consider this possibility. Is this something common? I've only tried with ~20 or so custom volumes.

4 and 5. As long as a volume exists, the prune volumes button is enabled. In my opinion, this should not be the case, but rather it should be enabled only when there is at least one volume that can be deleted. But given this, and my approach to "prune first just in case", we can't know whether one or more volumes are going to get deleted or not after pruning. That's why the method always returns true, either it finishes pruning and the number of volumes changes (which ends the loop) or finishes pruning without deleting anything, which after certain time we consider it has finished pruning and we stop waiting for it. 

The more I think about this the less I like it, so I'm going to check the status of the rows one by one, as it does not make sense to try to prune volumes if we know it's not going to do anything, and this way there is no awkward "wait for nothing to happen" in some cases.